`-Wdeprecated-non-prototype` warning in musl's `dlstart.c` with `zig cc main.c -target aarch64-linux-musl`

### Zig Version

0.10.0

### Steps to Reproduce and Observed Behavior

```
$ cat main.c
#include <stdio.h>

int main()
{
    printf("Hello World\n");
}
$ zig cc main.c -target aarch64-linux-musl
In file included from /home/alexrp/sysroot/lib/zig/lib/libc/musl/crt/rcrt1.c:3:
/home/alexrp/sysroot/lib/zig/lib/libc/musl/crt/../ldso/dlstart.c:146:20: warning: a function declaration without a prototype is deprecated in all versions of C and is treated as a zero-parameter prototype in C2x, conflicting with a subsequent definition [-Wdeprecated-non-prototype]
        GETFUNCSYM(&dls2, __dls2, base+dyn[DT_PLTGOT]);
                          ^
/home/alexrp/sysroot/lib/zig/lib/libc/musl/crt/rcrt1.c:11:13: note: conflicting prototype is here
hidden void __dls2(unsigned char *base, size_t *sp)
            ^
1 warning generated.
```

Also happens for `arm-linux-musleabihf`.

This did not occur in 0.9.1.

### Expected Behavior

No warning.

Got the same thing using zig as CC for a golang crossbuild:

```
CGO_ENABLED=1 GOOS=linux GOARCH=arm64 CC="zig cc -target aarch64-linux" CXX="zig c++ -target aarch64-linux" /opt/homebrew/bin/go build -tags ,crossbuild -trimpath -ldflags '-s -w -X sylr.dev/fix/cmd.Version=c3ca56c-dirty -extldflags ""' -o dist/fix-c3ca56c-dirty-linux-arm64
# sylr.dev/fix
In file included from /opt/homebrew/Cellar/zig/0.10.0/lib/zig/libc/musl/crt/rcrt1.c:3:
/opt/homebrew/Cellar/zig/0.10.0/lib/zig/libc/musl/crt/../ldso/dlstart.c:146:20: warning: a function declaration without a prototype is deprecated in all versions of C and is treated as a zero-parameter prototype in C2x, conflicting with a subsequent definition [-Wdeprecated-non-prototype]
        GETFUNCSYM(&dls2, __dls2, base+dyn[DT_PLTGOT]);
                          ^
/opt/homebrew/Cellar/zig/0.10.0/lib/zig/libc/musl/crt/rcrt1.c:11:13: note: conflicting prototype is here
hidden void __dls2(unsigned char *base, size_t *sp)
            ^
1 warning generated.
```

Reproducing steps:

```
mkdir -p ~/git && (cd ~/git && git clone https://github.com/sylr/fix)
cd git/fix
make crossbuild
```

--- a/src/musl.zig
+++ b/src/musl.zig
@@ -32,9 +32,6 @@ pub fn buildCRTFile(comp: *Compilation, crt_file: CRTFile, prog_node: std.Progre
         .crti_o => {
             var args = std.ArrayList([]const u8).init(arena);
             try addCcArgs(comp, arena, &args, false);
-            try args.appendSlice(&[_][]const u8{
-                "-Qunused-arguments",
-            });
             var files = [_]Compilation.CSourceFile{
                 .{
                     .src_path = try start_asm_path(comp, arena, "crti.s"),
@@ -47,9 +44,6 @@ pub fn buildCRTFile(comp: *Compilation, crt_file: CRTFile, prog_node: std.Progre
         .crtn_o => {
             var args = std.ArrayList([]const u8).init(arena);
             try addCcArgs(comp, arena, &args, false);
-            try args.appendSlice(&[_][]const u8{
-                "-Qunused-arguments",
-            });
             var files = [_]Compilation.CSourceFile{
                 .{
                     .src_path = try start_asm_path(comp, arena, "crtn.s"),
@@ -189,10 +183,6 @@ pub fn buildCRTFile(comp: *Compilation, crt_file: CRTFile, prog_node: std.Progre
 
                 var args = std.ArrayList([]const u8).init(arena);
                 try addCcArgs(comp, arena, &args, ext == .o3);
-                try args.appendSlice(&[_][]const u8{
-                    "-Qunused-arguments",
-                    "-w", // disable all warnings
-                });
                 const c_source_file = try c_source_files.addOne();
                 c_source_file.* = .{
                     .src_path = try comp.zig_lib_directory.join(arena, &[_][]const u8{ "libc", src_file }),
@@ -427,6 +417,9 @@ fn addCcArgs(
         "-fno-asynchronous-unwind-tables",
         "-ffunction-sections",
         "-fdata-sections",
+
+        "-Qunused-arguments",
+        "-w", // disable all warnings
     });
 }
 

